Could you please check as the test is failing on darwin?
Besides this, it looks fine to me.
<details><summary>Details</summary>
<p>

```
_________________________ TestPlugins.test_pixel_array _________________________
self = <pylibjpeg.tests.test_decode_pydicom.TestPlugins object at 0x104b86cd0>
    @pytest.mark.skipif(not RUN_JPEG, reason="No JPEG plugin")
    def test_pixel_array(self):
        # Should basically just not mess up the usual pydicom behaviour
>       index = get_indexed_datasets("1.2.840.10008.1.2.4.50")
pylibjpeg/tests/test_decode_pydicom.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/xifw15vh2xygglr0p01zd8dlpr1rd1dp-python3.11-pylibjpeg-data-1.0.0dev0/lib/python3.11/site-packages/ljdata/manager.py:81: in get_indexed_datasets
    index[fname]['ds'] = dcmread(fpath)
/nix/store/adn0vdldmqyl8lvjllbga17igw87508l-python3.11-pydicom-2.4.3/lib/python3.11/site-packages/pydicom/filereader.py:1030: in dcmread
    dataset = read_partial(
/nix/store/adn0vdldmqyl8lvjllbga17igw87508l-python3.11-pydicom-2.4.3/lib/python3.11/site-packages/pydicom/filereader.py:778: in read_partial
    preamble = read_preamble(fileobj, force)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
fp = <_io.BufferedReader name='/nix/store/xifw15vh2xygglr0p01zd8dlpr1rd1dp-python3.11-pylibjpeg-data-1.0.0dev0/lib/python3.11/site-packages/ljdata/ds/JPEGBaseline/color3d_jpeg_baseline.dcm'>
force = False
    def read_preamble(fp: BinaryIO, force: bool) -> Optional[bytes]:
        """Return the 128-byte DICOM preamble in `fp` if present.
    
        `fp` should be positioned at the start of the file-like. If the preamble
        and prefix are found then after reading `fp` will be positioned at the
        first byte after the prefix (byte offset 133). If either the preamble or
        prefix are missing and `force` is ``True`` then after reading `fp` will be
        positioned at the start of the file-like.
    
        Parameters
        ----------
        fp : file-like object
            The file-like to read the preamble from.
        force : bool
            Flag to force reading of a file even if no header is found.
    
        Returns
        -------
        preamble : bytes or None
            The 128-byte DICOM preamble will be returned if the appropriate prefix
            ('DICM') is found at byte offset 128. Returns ``None`` if the 'DICM'
            prefix is not found and `force` is ``True``.
    
        Raises
        ------
        InvalidDicomError
            If `force` is ``False`` and no appropriate header information found.
    
        Notes
        -----
        Also reads past the 'DICM' marker. Rewinds file to the beginning if
        no header found.
        """
        logger.debug("Reading File Meta Information preamble...")
        preamble = fp.read(128)
        if config.debugging:
            sample = bytes2hex(preamble[:8]) + "..." + bytes2hex(preamble[-8:])
            logger.debug(f"{fp.tell() - 128:08x}: {sample}")
    
        logger.debug("Reading File Meta Information prefix...")
        magic = fp.read(4)
        if magic != b"DICM" and force:
            logger.info(
                "File is not conformant with the DICOM File Format: 'DICM' "
                "prefix is missing from the File Meta Information header "
                "or the header itself is missing. Assuming no header and "
                "continuing."
            )
            fp.seek(0)
            return None
    
        if magic != b"DICM" and not force:
>           raise InvalidDicomError(
                "File is missing DICOM File Meta Information header or the 'DICM' "
                "prefix is missing from the header. Use force=True to force "
                "reading."
            )
E           pydicom.errors.InvalidDicomError: File is missing DICOM File Meta Information header or the 'DICM' prefix is missing from the header. Use force=True to force reading.
/nix/store/adn0vdldmqyl8lvjllbga17igw87508l-python3.11-pydicom-2.4.3/lib/python3.11/site-packages/pydicom/filereader.py:723: InvalidDicomError
```

</p>
</details>

--- a/pkgs/development/python-modules/pylibjpeg/default.nix
+++ b/pkgs/development/python-modules/pylibjpeg/default.nix
@@ -1,0 +1,69 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, setuptools
+, numpy
+, pydicom
+, pylibjpeg-libjpeg
+}:
+
+let
+  pylibjpeg-data = buildPythonPackage rec {
+    pname = "pylibjpeg-data";
+    version = "1.0.0dev0";
+    pyproject = true;
+
+    nativeBuildInputs = [ setuptools ];
+
+    src = fetchFromGitHub {
+      owner = "pydicom";
+      repo = "pylibjpeg-data";
+      rev = "2ab4b8a65b070656eca2582bd23197a3d01cdccd";
+      hash = "sha256-cFE1XjrqyGqwHCYGRucXK+q4k7ftUIbYwBw4WwIFtEc=";
+    };
+
+    doCheck = false;
+  };
+in
+
+buildPythonPackage rec {
+  pname = "pylibjpeg";
+  version = "1.4.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "pydicom";
+    repo = "pylibjpeg";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-Px1DyYDkKAUdYo+ZxZ1w7TkPzWN++styiFl02iQOvyQ=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [ numpy ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pydicom
+    pylibjpeg-data
+    pylibjpeg-libjpeg
+  ];
+
+  pythonImportsCheck = [ "pylibjpeg" ];
+
+  meta = with lib; {
+    description = "Python framework for decoding JPEG images, with a focus on supporting Pydicom";
+    homepage = "https://github.com/pydicom/pylibjpeg";
+    changelog = "https://github.com/pydicom/pylibjpeg/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ bcdarwin ];
+    # several test failures of form
+    # "pydicom.errors.InvalidDicomError: File is missing DICOM File Meta Information header or the 'DICM' prefix is missing from the header. ..."
+    broken = stdenv.isDarwin;
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -10602,6 +10602,8 @@ self: super: with self; {
     inherit (pkgs) libusb1;
   };
 
+  pylibjpeg = callPackage ../development/python-modules/pylibjpeg { };
+
   pylibjpeg-libjpeg = callPackage ../development/python-modules/pylibjpeg-libjpeg { };
 
   pyliblo = callPackage ../development/python-modules/pyliblo { };

